collectorist/api/src/db/mod.rs

```diff
@@ -183,6 +192,54 @@ impl Db {
         })
     }
 
+    async fn filter_purls_as_of(
+        &self,
+        collector_id: &str,
+        input: Vec<String>,
+        as_of: DateTime<Utc>,
+    ) -> Result<Vec<String>, anyhow::Error> {
+        // Per https://stackoverflow.com/questions/70029671/how-to-query-using-an-in-clause-and-a-vec-as-parameter-in-rust-sqlx-for-mysql
```

Comment:
cool I also do this kind of thing ( adding external URLs )

Reply:
Nothing magic. Just external explanation as to what method I'm following, and why.